src/tools/cephfs/top/cephfs-top

```diff
@@ -591,10 +666,75 @@ class FSTop(object):
-                    self.refresh_main_window(x_coord_map, stats_json)
+                    self.refresh_main_window(x_coord_map, stats_json, fs)
+                self.exit_ev.wait(timeout=self.refresh_interval_secs)
+                curses.halfdelay(self.refresh_interval_secs)
+                c = self.header.getch()
+                self.set_option(c)
+            except curses.error:
+                # this may happen when addstr the terminal window
+                # size changed, just retry it
+                pass
+
+    def run_all_display(self, fs_list):
+        while not self.exit_ev.is_set():
+            # use stdscr.clear() instead of clearing each window
+            # to avoid screen blinking.
+            self.stdscr.clear()
+            self.refresh_window_size()
+            if self.height <= 4 or self.width <= 2:
+                self.exit_ev.wait(timeout=self.refresh_interval_secs)
+                continue
+
+            # coordinate constants for windowing -- (height, width, y, x)
+            # NOTE: requires initscr() call before accessing COLS, LINES.
+            try:
+                HEADER_WINDOW_COORD = (4, self.width - 1, 0, 0)
+                self.header = curses.newwin(*HEADER_WINDOW_COORD)
+            except curses.error:
+                # this may happen when creating the sub windows the
+                # terminal window size changed, just retry it
+                continue
+
+            stats_json = self.perf_stats_query()
+            try:
+                main, top, num_client, self.flag = 7, 4, 0, 1
+                for fs in fs_list:
+                    fs_found = False
+                    for _ in range(0, 5):
+                        if not stats_json[CLIENT_METADATA_KEY]:
+                            time.sleep(1)
+                            stats_json = self.perf_stats_query()
+                        else:
+                            fs_found = True
+                            break
+                    if fs_found is False:
+                        raise FSTopException("Data is not present")
+                    if self.height >= 5 + num_client:
+                        TOPLINE_WINDOW_COORD = (3, self.width - 1, top, 0)
+                        self.topl = curses.newwin(*TOPLINE_WINDOW_COORD)
+                    else:
+                        self.topl = None
+                    if self.height >= 8 + num_client:
+                        MAIN_WINDOW_COORD = (self.height - 7, self.width - 1,
+                                             main, 0)
+                        self.mainw = curses.newwin(*MAIN_WINDOW_COORD)
+                    else:
+                        self.mainw = None
+                    num_client = len(stats_json[CLIENT_METADATA_KEY][fs])
+                    top = top + (num_client + 4)
+                    main = main + (num_client + 4)
+                    headline = "Headline: All Filesystem info [Press 'm' to select a filesystem!]"
```

Comment:
Suggestion: I feel `!` could be removed, just "Press 'q' to see all filesystems" looks better.